rollup.config.js

```diff
@@ -0,0 +1,173 @@
+import fs from 'fs';
```

Comment:
We'll need to make this a major version change, right?

Reply:
how so? dependees don't need to change any code and nothing should break for them.

Reply:
The resulting artifact is produced through an entirely different build system. Can you guarantee it'll cause no regressions for applications?

tl;dr: I think it's safer and easier to make this a breaking change for all packages.

Reply:
It sounds like you want to use major version to communicate "this should be tested more thoroughly by your app before upgrading", but that's not really what major version means if we follow semver. per semver it just means "the api changed so review the documented changes and if you update your usage accordingly then things should just work". 
i.e. the version number only indicates _intentional_ breakages, not the risk-level associated with unintentional breakages.

I'm ok with wanting version numbers' meaning to extend beyond the semver contract, but I want to be explicit about that, especially since this is not documented. Our readme merely states:
>We follow semver conventions for versioning, so each change will either be major, minor, or patch.

which is not entirely accurate if we're extending the definition of major version to also mean "higher likelihood of undocumented breakages so test more thoroughly".

Reply:
What would you suggest we release this as?

Reply:
If you consider adding ESM module support as a feature, then following semver this would be a minor release

Reply:
My read of semver is that it's not strict with when a major version is released.

[Line 8 says](https://semver.org/#spec-item-8)
> Major version X (X.y.z | X > 0) MUST be incremented if any backwards incompatible changes are introduced to the public API. It MAY also include minor and patch level changes. Patch and minor version MUST be reset to 0 when major version is incremented.

Reply:
I also want @DesignerDave to gain consensus on this as well, so ping on that.

Reply:
We've done major versions for similar changes where we think there's a high likelihood of something breaking down the line. By my understanding of these changes, we're no longer building the `index.bundle.js` files with webpack, which will cause very different output builds, so I think this is a change with a high-risk of breaking consuming applications. I'm definitely with Harry on this, and think we should be conservative with our versioning here, and in general.